xcode/Subconscious/Shared/Services/DataService.swift

```diff
@@ -154,42 +154,67 @@ actor DataService {
         try database.savepoint(savepoint)
         
         do {
+            var toUpdate: [(Slug, Memo)] = []
+            var toRemove: [Slug] = []
```

Comment:
First pass: fetch all changes from Noosphere
Second pass: apply all changes to index